dev/src/document-change.ts

```diff
@@ -36,7 +41,9 @@ export class DocumentChange {
    * @param {number} newIndex - The index in the documents array after this
    * change.
    */
-  constructor(type, document, oldIndex, newIndex) {
+  constructor(
+      type: string, document: QueryDocumentSnapshot, oldIndex: number,
+      newIndex: number) {
```

Comment:
Commenting on the methods below: Should we add the return type annotation?

Reply:
I tend to only specify the type annotation when it can't be inferred.  In almost all of these cases, it can be.